### PR TITLE
XPath parser and evaluator

### PR DESCRIPTION
**Pull request by funkyfuture**
this is the little efforts i already put into an XPath parser b/c @JKatzwinkel was already nagging about details.
a first conclusion is that the primary use of `pegen` as Python source parser requires some hacks, but no nasty ones; i'm still optimistic to rely on it.

let's try to keep this thread tidy, i will henceforth delete resolved topics (please try address one topic per comment, if possible) and integrate the results in the following TODO list. for complicated / general design issues separate issues are fine too.

- [ ] remove the generated file from vcs
  - shall type checks include this file?
- [ ] document the deviations from https://www.w3.org/TR/1999/REC-xpath-19991116/
- [ ] check that all `TODO`-like annotations in this changeset have been solved
- [ ] update related item in the FAQ section

### CODE REVIEW

**Comment by JKatzwinkel** [comment]
caring about codebase consistency and noise reduction === nagging got it! 📝

File: _delb/nodes.py [hunk 1]
```diff
@@ -56,6 +56,7 @@
     _random_unused_prefix,
 )
 from _delb.xpath import LegacyLocationStep, LegacyXPathExpression
+from _delb.xpath import evaluate
```
Comment (funkyfuture) [inline _delb/nodes.py:59]: please change to `… as evaluate_xpath`.

**Comment by funkyfuture** [comment]
let's continue this by evaluating https://github.com/we-like-parsers/pegen/pull/65

**Comment by funkyfuture** [comment]
This approach is replaced by #55.

**Pull request closed without merging**